mlflow/models/evaluation/default_evaluator.py

```diff
@@ -332,6 +349,37 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _infer_artifact_type_and_ext(artifact_name, raw_artifact, exception_header):
```

Comment:
This function is all about artifact type inference. I suggest to move it to mlflow/models/evaluation/artifacts.py instead.